src/common/base/ConcurrentLRUCache.h

```diff
@@ -206,19 +206,21 @@ class LRU {
     }
 
     void insert(key_type&& key, value_type&& value) {
-        typename map_type::iterator i = map_.find(key);
-        if (i == map_.end()) {
-            // insert item into the cache, but first check if it is full
-            if (size() >= capacity_) {
-                VLOG(3) << "Size:" << size() << ", capacity " << capacity_;
-                // cache is full, evict the least recently used item
-                evict();
-            }
-            VLOG(3) << "Insert key " << key << ", val " << value;
-            // insert the new item
+        // insert item into the cache, but first check if it is full
+        if (size() >= capacity_) {
+            VLOG(3) << "Size:" << size() << ", capacity " << capacity_;
+            // cache is full, evict the least recently used item
+            evict();
+        }
+        VLOG(3) << "Insert key " << key << ", val " << value;
+        typename map_type::iterator it = map_.find(key);
+        if (it == map_.end()) {
             list_.push_front(key);
             map_.emplace(std::forward<key_type>(key),
                          std::make_tuple(std::forward<value_type>(value), list_.begin()));
+        } else {
+            // Overwrite the value
+            std::get<0>(it->second) = std::move(value);
```

Comment:
also need to move this value to front of list ?

Reply:
If so, the operation becomes evict the old key then insert the new key.  
I am not sure which way is better.

Reply:
👍

Reply:
Make sense